### PR TITLE
[Case 6491] Part 2 of entitySelectionTool clean up

### PR DESCRIPTION
**Pull request by 1P-Cusack**
Part 2 of [FogBugz Case 6491](https://highfidelity.fogbugz.com/f/cases/6491/edit-js-work-proposed-by-1st-playable);  part 1 was merged via PR #11165.  The overall PR for the case is broken into preview chunks to allow for review in smaller logical pieces.

This PR matches the content of [Case 6491: Preview PR 1](https://github.com/1P-Cusack/hifi/pull/2) mentioned in pull request #11165

The current work in addition to this phase may be viewed through several preview pull requests located:
[Case 6491: Preview PR 2](https://github.com/1P-Cusack/hifi/pull/3)
[Case 6491: Preview PR 3](https://github.com/1P-Cusack/hifi/pull/4)
[Case 6491: Preview PR 4](https://github.com/1P-Cusack/hifi/pull/5)
[Case 6491: Preview PR 5](https://github.com/1P-Cusack/hifi/pull/6)
[Case 6491: Preview PR 6](https://github.com/1P-Cusack/hifi/pull/8)

Description:
This fixes the issue where grabbers would re-appear when rotating
the selected object rather than staying hidden.

updateHandles will now take the mode into account when deciding
if the non-light grabber handles should be visible.  This can be
subverted by the user selecting a light source as in line with
the previous behavior.

Adds some debug prints in event handlers guarded by local
wantDebug checks.

Has some minor cleanup changes:
* Remove unused var within updateRotationHandles
* Readability improvement for light check within updateHandles
* Pulled up rotate mode check within updateHandles
* Added grabberCloner visibility flag within updateHandles

Testing:

- Clicking on any of the rotation handles (pitch, yaw, roll) will have all grabbers on the selection go invisible.
- Actually rotating will see all the grabbers remain invisible, until the user stops rotating and releases the mouse press.
- When the user releases the mouse press, all grabbers will become visible once again.

NOTE:  The degree snapping during rotation is a known issue and will be addressed in a subsequent pull request.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2574/HighFidelity-Beta-PR11205-7de69710df05a2a2fa1f672a96485ee5e3e1ae68.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2574/HighFidelity-Beta-PR11205-7de69710df05a2a2fa1f672a96485ee5e3e1ae68.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

Clicking on any of the rotation handles (pitch, yaw, roll) will have all grabbers on the selection go invisible. **Fail, unless intended - the vertical grabber stays visible, please see image.**
image

Actually rotating will see all the grabbers remain invisible, until the user stops rotating and releases the mouse press. **Pass**
When the user releases the mouse press, all grabbers will be come visible once again. **Pass**

**Comment by 1P-Cusack** [comment]
I've not been able to repro the behavior that you're seeing with vertical translation handle on a local build.  I grabbed the windows build from the link above; however, I wasn't able to repro unable the issue there either.  On both builds, the handles behaved as expected.

Are there any other particular steps to reproduce this issue?

**Comment by YuppyPlays** [comment]
@1P-Cusack  
Not as far as I can tell, just went into interfaces Home location and used Create for a cube, then started using the far-grab lasers on the created cubes grabbers.

**Comment by 1P-Cusack** [comment]
Do you get the same behavior with your HMD off, using the mouse?  I don't see the rotation overlays that should be present when actively rotating.

**Comment by YuppyPlays** [comment]
@1P-Cusack  Seems so yeah, see pic below of during dragging the grabber up/down:
image

**Comment by 1P-Cusack** [comment]
I'm not having any fortune reproducing this issue, either on a local build or on the [PR 11205 Autobuild](https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2574/HighFidelity-Beta-PR11205-7de69710df05a2a2fa1f672a96485ee5e3e1ae68.exe).

To get an additional data point, can someone else test this?  Can anyone else think of what differences may trigger this behavior?  I thought that perhaps there was a passive merge error; however, as mentioned before I wasn't able to reproduce the issue on the autobuild executable.

**Comment by cain-kilgore** [comment]
@1P-Cusack I'll give this a shot shortly and let you know the results.

**Comment by cain-kilgore** [comment]
@1P-Cusack 

Hopefully this video shows off what's going on here with the Vertical Grabber.
https://s3-eu-west-1.amazonaws.com/cainkilgore-prod/files/e6b89ff6-723c-4ee3-9cbf-1d562cf1c020.webm

**Comment by 1P-Cusack** [comment]
@cain-kilgore:   Thanks for the video, it was very useful!  I took a look at it and I believe that there may be some miscommunication over what the expected results for the vertical arrow are.  The expected results listed here are for choosing to rotate and actively _rotating_, which is the feature that this PR is concentrating on.  The vertical arrow (the translateY grabber graphic) is a different action than rotation and has a different visual cue to go along with it.

The rotation action when initiated and while in progress, should have all of the grabbers (rotation, translation, scale/sizing related) be invisible until the next mouse release event is triggered.  While in progress, its visual cue, the rotation overlay ui, should be visible.  That's the two concentric circles and label with the degrees on it. 

The translateY action is handled in a subsequent pr; however, until this it should retain the same behavior as it does on master.  While in progress the translationY visual cue is its handle which remains visible and in a highlighted state and _only_ the rotation handles (pitch, yaw, roll) should be invisible.

I've taken some snapshots from the video you provided to show the steps for rotation.  Looking at the video things are working according to the expected results.  Let me know if there are any questions regarding the notations listed in the screenshots or if there's some behavior that you've seen that I may have missed when reviewing the video.

**Comment by cain-kilgore** [comment]
@1P-Cusack: Ah, the Y Rotation - that would explain it. In that case this is all good for me.

**Comment by 1P-Cusack** [comment]
@cain-kilgore:  Ok, thanks!  If this is QA Approved, then the next step would be getting this pull request code reviewed.  I'm not sure how to trigger that process for a FogBugz Case.  If this were a Worklist item, I could switch the status to Code Review.  Is there something similar for cases?  @themelissabrown:  Is there a preferred process for this or does a comment notation within the case suffice in situations where the fix involves multiple pull requests?

**Comment by themelissabrown** [comment]
@1P-Cusack will get a review on this now

File: scripts/system/libraries/entitySelectionTool.js [hunk 1]
```diff
@@ -3547,30 +3570,43 @@ SelectionDisplay = (function() {
...
             y: Math.sin(angle) * outerRadius * ROTATION_DISPLAY_DISTANCE_MULTIPLIER,
             z: 0,
         };
+        print("    Angle: " + angle );
+        print("    InitialPos: " + position.x + ", " + position.y + ", " + position.z);
         position = Vec3.multiplyQbyV(handleRotation, position);
         position = Vec3.sum(centerPosition, position);
-        Overlays.editOverlay(rotationDegreesDisplay, {
+        print("    TranslatedPos: " + position.x + ", " + position.y + ", " + position.z);
+        var overlayProps = {
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: 1.2.12 - Coding standards suggest that this not be abbreviated

File: scripts/system/libraries/entitySelectionTool.js [hunk 2]
```diff
@@ -4123,47 +4192,74 @@ SelectionDisplay = (function() {
                     case grabberNEAR:
                     case grabberEdgeTN: // TODO: maybe this should be TOP+NEAR stretching?
                     case grabberEdgeBN: // TODO: maybe this should be BOTTOM+FAR stretching?
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 3]
```diff
@@ -4123,47 +4192,74 @@ SelectionDisplay = (function() {
...
+                            print("grabberNear variant");
+                        }
                         mode = "STRETCH_NEAR";
                         somethingClicked = mode;
                         break;
 
                     case grabberFAR:
                     case grabberEdgeTF: // TODO: maybe this should be TOP+FAR stretching?
                     case grabberEdgeBF: // TODO: maybe this should be BOTTOM+FAR stretching?
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 4]
```diff
@@ -4123,47 +4192,74 @@ SelectionDisplay = (function() {
...
+                            print("grabberFar variant");
+                        }
                         mode = "STRETCH_FAR";
                         somethingClicked = mode;
                         break;
                     case grabberTOP:
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 5]
```diff
@@ -4123,47 +4192,74 @@ SelectionDisplay = (function() {
...
+                            print("grabberTOP");
+                        }
                         mode = "STRETCH_TOP";
                         somethingClicked = mode;
                         break;
                     case grabberBOTTOM:
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 6]
```diff
@@ -4123,47 +4192,74 @@ SelectionDisplay = (function() {
...
+                            print("grabberBOTTOM");
+                        }
                         mode = "STRETCH_BOTTOM";
                         somethingClicked = mode;
                         break;
                     case grabberRIGHT:
                     case grabberEdgeTR: // TODO: maybe this should be TOP+RIGHT stretching?
                     case grabberEdgeBR: // TODO: maybe this should be BOTTOM+RIGHT stretching?
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 7]
```diff
@@ -4123,47 +4192,74 @@ SelectionDisplay = (function() {
...
+                            print("grabberRight variant");
+                        }
                         mode = "STRETCH_RIGHT";
                         somethingClicked = mode;
                         break;
                     case grabberLEFT:
                     case grabberEdgeTL: // TODO: maybe this should be TOP+LEFT stretching?
                     case grabberEdgeBL: // TODO: maybe this should be BOTTOM+LEFT stretching?
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 8]
```diff
@@ -4123,47 +4192,74 @@ SelectionDisplay = (function() {
...
+                            print("grabberLeft variant");
+                        }
                         mode = "STRETCH_LEFT";
                         somethingClicked = mode;
                         break;
 
                     default:
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 9]
```diff
@@ -4123,47 +4192,74 @@ SelectionDisplay = (function() {
...
+                            print("UNKNOWN( " + result.overlayID + " )");
+                        }
                         mode = "UNKNOWN";
                         break;
                 }
+                if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 10]
```diff
@@ -4123,47 +4192,74 @@ SelectionDisplay = (function() {
...
+                    print("  Unregistered Tool Mode: " + mode );
+                }
             }
         }
 
         // if one of the items above was clicked, then we know we are in translate or stretch mode, and we
         // should hide our rotate handles...
         if (somethingClicked) {
+            if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 11]
```diff
@@ -4175,6 +4271,9 @@ SelectionDisplay = (function() {
             });
 
             if (mode != "TRANSLATE_UP_DOWN") {
+                if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 12]
```diff
@@ -4207,41 +4306,70 @@ SelectionDisplay = (function() {
 
             if (result.intersects) {
                 var resultTool = grabberTools[result.overlayID];
+                if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 13]
```diff
@@ -4207,41 +4306,70 @@ SelectionDisplay = (function() {
...
+                    print("Intersection detected with handle...");
+                }
                 if (resultTool) {
+                    if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 14]
```diff
@@ -4207,41 +4306,70 @@ SelectionDisplay = (function() {
...
+                        print("    " + resultTool.mode);
+                    }
                     activeTool = resultTool;
                     mode = resultTool.mode;
                     somethingClicked = 'tool';
                     if (activeTool && activeTool.onBegin) {
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 15]
```diff
@@ -4207,41 +4306,70 @@ SelectionDisplay = (function() {
...
+                            print("    Triggering Tool's onBegin");
+                        }
                         activeTool.onBegin(event);
+                    } else if(wantDebug) {
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if

File: scripts/system/libraries/entitySelectionTool.js [hunk 16]
```diff
@@ -4207,41 +4306,70 @@ SelectionDisplay = (function() {
...
+                        print("    Tool's missing onBegin");
                     }
                 }
                 switch (result.overlayID) {
                     case yawHandle:
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 17]
```diff
@@ -4207,41 +4306,70 @@ SelectionDisplay = (function() {
...
+                            print("Handle_YAW");
+                        }
                         mode = "ROTATE_YAW";
                         somethingClicked = mode;
                         overlayOrientation = yawHandleRotation;
                         overlayCenter = yawCenter;
                         yawZero = result.intersection;
                         rotationNormal = yawNormal;
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 18]
```diff
@@ -4207,41 +4306,70 @@ SelectionDisplay = (function() {
...
+                            print("rotationNormal set to: " + rotationNormal.x + ", " + rotationNormal.y + ", " + rotationNormal.z);
+                        }
                         break;
 
                     case pitchHandle:
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 19]
```diff
@@ -4207,41 +4306,70 @@ SelectionDisplay = (function() {
...
+                            print("Handle_PITCH");
+                        }
                         mode = "ROTATE_PITCH";
                         initialPosition = SelectionManager.worldPosition;
                         somethingClicked = mode;
                         overlayOrientation = pitchHandleRotation;
                         overlayCenter = pitchCenter;
                         pitchZero = result.intersection;
                         rotationNormal = pitchNormal;
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 20]
```diff
@@ -4207,41 +4306,70 @@ SelectionDisplay = (function() {
...
+                            print("rotationNormal set to: " + rotationNormal.x + ", " + rotationNormal.y + ", " + rotationNormal.z);
+                        }
                         break;
 
                     case rollHandle:
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 21]
```diff
@@ -4207,41 +4306,70 @@ SelectionDisplay = (function() {
...
+                            print("Handle_ROLL");
+                        }
                         mode = "ROTATE_ROLL";
                         somethingClicked = mode;
                         overlayOrientation = rollHandleRotation;
                         overlayCenter = rollCenter;
                         rollZero = result.intersection;
                         rotationNormal = rollNormal;
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 22]
```diff
@@ -4259,6 +4387,9 @@ SelectionDisplay = (function() {
 
             if (somethingClicked) {
 
+                if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 23]
```diff
@@ -4283,6 +4414,9 @@ SelectionDisplay = (function() {
                     startAt: 0,
                     endAt: 0
                 });
+                if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 24]
```diff
@@ -4293,15 +4427,11 @@ SelectionDisplay = (function() {
                     visible: false
                 });
 
+                if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 25]
```diff
@@ -4397,6 +4527,9 @@ SelectionDisplay = (function() {
                 switch (result.overlayID) {
                     case selectionBox:
                         activeTool = translateXZTool;
+                        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 26]
```diff
@@ -4430,7 +4563,10 @@ SelectionDisplay = (function() {
         }
 
         // reset everything as intersectable...
-        // TODO: we could optimize this since some of these were already flipped back
+        // TODO: we could optimize this since some of these were already flipped back(i.e:  just get rid of this)
+        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 27]
```diff
@@ -4444,14 +4580,34 @@ SelectionDisplay = (function() {
...
+            print( "=============== eST::MousePressEvent END =======================");
+        }
+
         return somethingClicked;
     };
 
     // FUNCTION: MOUSE MOVE EVENT
     that.mouseMoveEvent = function(event) {
+        var wantDebug = false;
+        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 28]
```diff
@@ -4444,14 +4580,34 @@ SelectionDisplay = (function() {
...
+            print( "=============== eST::MouseMoveEvent BEG =======================");
+        }
         if (activeTool) {
+            if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 29]
```diff
@@ -4444,14 +4580,34 @@ SelectionDisplay = (function() {
...
+                print("    Trigger ActiveTool( " + activeTool.mode + " )'s onMove");
+            }
             activeTool.onMove(event);
+            
+            if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 30]
```diff
@@ -4444,14 +4580,34 @@ SelectionDisplay = (function() {
...
+                print("    Trigger SelectionManager::update");
+            }
             SelectionManager._update();
+            
+            if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 31]
```diff
@@ -4574,6 +4730,9 @@ SelectionDisplay = (function() {
             }
         }
 
+        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 32]
```diff
@@ -4618,13 +4777,27 @@ SelectionDisplay = (function() {
 
     // FUNCTION: MOUSE RELEASE EVENT
     that.mouseReleaseEvent = function(event) {
+        var wantDebug = false;
+        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 33]
```diff
@@ -4618,13 +4777,27 @@ SelectionDisplay = (function() {
...
+            print("=============== eST::MouseReleaseEvent BEG =======================");
+        }
         var showHandles = false;
-        if (activeTool && activeTool.onEnd) {
-            activeTool.onEnd(event);
+        if (activeTool) {
+            if( activeTool.onEnd ) {
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if

File: scripts/system/libraries/entitySelectionTool.js [hunk 34]
```diff
@@ -4618,13 +4777,27 @@ SelectionDisplay = (function() {
...
+                if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 35]
```diff
@@ -4618,13 +4777,27 @@ SelectionDisplay = (function() {
...
+                    print("    Triggering ActiveTool( " + activeTool.mode + " )'s onEnd");
+                }
+                activeTool.onEnd(event);
+            }else if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space before else, after if, before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 36]
```diff
@@ -4618,13 +4777,27 @@ SelectionDisplay = (function() {
...
+                print("    ActiveTool( " + activeTool.mode + " )'s missing onEnd");
+            }
         }
-        activeTool = null;
+        
         // hide our rotation overlays..., and show our handles
         if (mode == "ROTATE_YAW" || mode == "ROTATE_PITCH" || mode == "ROTATE_ROLL") {
+            if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 37]
```diff
@@ -4637,22 +4810,26 @@ SelectionDisplay = (function() {
...
         }
 
+        showHandles = (mode != "UNKNOWN");//<base on prior mode
         mode = "UNKNOWN";
+        activeTool = null;
 
         // if something is selected, then reset the "original" properties for any potential next click+move operation
         if (SelectionManager.hasSelection()) {
             if (showHandles) {
+                if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

File: scripts/system/libraries/entitySelectionTool.js [hunk 38]
```diff
@@ -4637,22 +4810,26 @@ SelectionDisplay = (function() {
...
+                    print("    Triggering that.select");
+                }
                 that.select(SelectionManager.selections[0], event);
             }
         }
 
+        if(wantDebug){
```
Comment (misslivirose) [inline scripts/system/libraries/entitySelectionTool.js]: space after if and before {

**Comment by 1P-Cusack** [comment]
@misslivirose:  Thanks for the review!  This pull request in one of several, and I've another round of changes that are inline after this pull request which address the majority, if not all, of the style changes regarding the debug statements you mention here.  That change list has some other functionality needed, as this request is one in the line of several.  If you and @themelissabrown are okay with approving this, then I can get those changes ([Case 6491: Preview PR 2](https://github.com/1P-Cusack/hifi/pull/3)) in line for QA & code review once this is merged.

https://github.com/highfidelity/hifi/pull/11205#issue-250985766 has the links of the awaiting changes.  When all the preview pull requests are in the debug statements should all be under guard, with any warnings or error messages the only things that print out to console in the event of an issue; however, until then they'll come in phases of each pull request.

**Comment by 1P-Cusack** [comment]
@misslivirose:  I didn't know if you have the chance to see my last comment: https://github.com/highfidelity/hifi/pull/11205#issuecomment-327542728, so just giving this thread a bump.  I'd originally broke up the pr request for ease of review as it was quite large; however, I'll be uploading the rest of the changes in a larger pr which will have these issues addressed.  Is it okay to have this approved and land in the build, as the wantDebug spacing checks are addressed in the upcoming pull request?

**Comment by misslivirose** [comment]
If the changes are going to be overwritten by an incoming PR, I'd rather review that PR even with a larger diff than approve an in-progress one as it will kick off a build and update the dev domains. It sounds like closing this PR and submitting the next one might work best.

**Comment by 1P-Cusack** [comment]
@themelissabrown :  Given the discussion with @misslivirose, I can either update this pr with the remainder of the changes, as all the sectioned pull request build on one another, or this pr can be closed out and a new pull request can be opened which will have these changes & the remaining changes incorporated.  There will need to be another QA pass either way, as it'll bring in a number of functionality changes in addition to what's currently here, so additional testing steps will need to be added.   Let me know which is preferred and I'll get to it as soon as I can.

**Comment by 1P-Cusack** [comment]
Closing this pull request, as discussed with @misslivirose & @themelissabrown, in favor of requested larger pull request with the remainder of changes, including that covered by this pull request.

**Pull request closed without merging**